execution/src/execution.cc

```diff
@@ -350,13 +350,22 @@ namespace leatherman { namespace execution {
         return execute(file, &arguments, nullptr, &environment, nullptr, stdout_callback, stderr_callback, actual_options, timeout).success;
     }
 
-    static bool process_data(bool trim, string const& data, string& buffer, string const& logger, function<bool(string&)> const& callback)
+    static bool process_data(bool trim, bool strip_cr, string const& raw_data, string& buffer, string const& logger, function<bool(string&)> const& callback)
     {
         // Do nothing if nothing was read
-        if (data.empty()) {
+        if (raw_data.empty()) {
             return true;
         }
 
+        string stripped_data;
```

Comment:
I believe that if I made this static, we would avoid having to allocate every loop (at some point it would just end up at the output buffer size, and continue to be overwritten in that location). The tradeoff there would be that we'd leak that 4K, as there's no object to tie the lifetime of this variable to.

A third option is to make the series of `data` arguments eventually leading to this function non-const. That would allow us to modify the buffer in place.

Yet a fourth would be to try moving this logic to the `rw_from_child` method in windows.cc, which may actually be the best option - it keeps this windows-focused code in the right place, and avoids the need to change the "public" API mentioned above